tsweb/tsweb_test.go

```diff
@@ -485,8 +485,16 @@ func TestStdHandler(t *testing.T) {
 				Step:  time.Second,
 			})
 
+			onStartRecord := &AccessLogRecord{}
+			onCompletionRecord := &AccessLogRecord{}
 			rec := noopHijacker{httptest.NewRecorder(), false}
-			h := StdHandler(test.rh, HandlerOptions{Logf: logf, Now: clock.Now, OnError: test.errHandler})
+			h := StdHandler(test.rh, HandlerOptions{
+				Logf:         logf,
+				Now:          clock.Now,
+				OnError:      test.errHandler,
+				OnStart:      func(r *http.Request, alr AccessLogRecord) { *onStartRecord = alr },
+				OnCompletion: func(r *http.Request, alr AccessLogRecord) { *onCompletionRecord = alr },
```

Comment:
are these pointers or values? why the mix?

Reply:
Not sure I follow - existing `OnCompletion` passed `AccessLogRecord` by value, so I've added `OnStart` to do the same. It's not a big struct, and `tsweb` continues to mutate it after calling these hooks, so I think passing by value makes sense, but I am happy to change it.

Reply:
why are they pointers on line 488 and 489? why not just values there?

Reply:
Oh right, these were left from a slightly different version of the test. Changed to values.